Python/bytecodes.c

```diff
@@ -279,73 +279,111 @@ dummy_func(
 
         family(binary_op, INLINE_CACHE_ENTRIES_BINARY_OP) = {
             BINARY_OP,
-            BINARY_OP_ADD_FLOAT,
+            BINARY_OP_MULTIPLY_INT,
             BINARY_OP_ADD_INT,
-            BINARY_OP_ADD_UNICODE,
-            // BINARY_OP_INPLACE_ADD_UNICODE,  // This is an odd duck.
+            BINARY_OP_SUBTRACT_INT,
             BINARY_OP_MULTIPLY_FLOAT,
-            BINARY_OP_MULTIPLY_INT,
+            BINARY_OP_ADD_FLOAT,
             BINARY_OP_SUBTRACT_FLOAT,
-            BINARY_OP_SUBTRACT_INT,
+            BINARY_OP_ADD_UNICODE,
+            // BINARY_OP_INPLACE_ADD_UNICODE,  // See comments at that opcode.
         };
 
-
-        inst(BINARY_OP_MULTIPLY_INT, (unused/1, left, right -- prod)) {
+        op(_BINARY_OP_INT_GUARD, (left, right -- left, right)) {
```

Comment:
Also, I think maybe the name could just be `_GUARD_INT` or something, since this could potentially be reused for things like `COMPARE_OP`.

Reply:
I'll use `_GUARD_BOTH_INT`, `_GUARD_BOTH_FLOAT`, and `_GUARD_BOTH_UNICODE`.